crates/core/tedge_mapper/src/core/error.rs

```diff
@@ -109,4 +110,7 @@ pub enum ConversionError {
 
     #[error("Failed to extract the child device name from file path : {dir}")]
     DirPathComponentError { dir: PathBuf },
+
+    #[error(transparent)]
+    FromC8YRestError(#[from] C8YRestError),
```

Comment:
Still here - but okay the issue being beyond the scope of this PR. The `core::ConversionError`  is already a mix of unrelated errors. This will have to be fix in a follow-up task.

Reply:
Oops, missed it. Will do it, as I'll have to resolve this anyway as part of moving the c8y-mapper into `extensions/c8y_mapper_ext`.

Reply:
I ended up moving the `Converter` trait and `ConversionError` enum from `core` to `c8y_mapper_ext` as no other crate was using it so it didn't make sense to keep them as separate `core` abstractions.